### PR TITLE
add restore purchases for ios

### LINKED ISSUE #16723: restore purchases on iOS
**Godot version:**
2.1.4

**OS/device including version:**
iOS

**Issue description:**
iOS needs a way to restore purchases.

Some types of in app purchases can be restored (for example, if the user moves to a different device).

### PR DESCRIPTION
**Pull request by jandrewlong**
Adds a restore_purchases function to InAppStore

Fixes #16723

### CODE REVIEW

File: platform/iphone/in_app_store.mm [hunk 1]
```diff
@@ -153,6 +154,14 @@ - (void)productsRequest:(SKProductsRequest *)request didReceiveResponse:(SKProdu
 	return OK;
 };
 
+Error InAppStore::restore_purchases() {
```
[the remark this answers is not in the document]
Comment (jandrewlong) [reply platform/iphone/in_app_store.mm]: It's not needed, so I removed it from the header.

(I had copy/pasted the line, and forgot to delete the argument)

**Comment by de11ed** [comment]
What a lifesaver! Any way I could get this working in 3.0.1?

**Comment by de11ed** [comment]
I see, that would be awesome!

**Comment by hpvb** [comment]
Cherrypicked into 3.0.3

**Comment by de11ed** [comment]
Anyone want to give me a quick guide/ example on how to use this? :)

**Pull request merged**